src/repository/transport/fetch.transport.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * Fetch transport to make HTTP requests
+ */
+export default class FetchTransport {
+  /**
+   * Fetch constructor
+   *
+   * @param baseUrl - Base URL
+   */
+  constructor(private readonly baseUrl: string) {
+  }
+
+  /**
+   * Make GET request
+   *
+   * @template Response - Response data type
+   * @param endpoint - API endpoint
+   * @param headers - Request headers
+   * @returns { Promise<Response> } - Response data
+   */
+  public async get<Response>(endpoint: string, headers: Record<string, string>): Promise<Response> {
+    // eslint-disable-next-line no-undef
+    const response = await fetch(this.baseUrl + endpoint, {
+      method: 'GET',
+      headers,
```

Comment:
are you sure that simple `Record` will work here? now a `new Headers()` ? do we need to convert `headers` argument to the `Headers()` set?

Reply:
It will work, i think, because we can pass to headers `HeadersInit` type, which includes `Record<string, string>`